src/components/Combobox.tsx

```diff
@@ -36,15 +57,23 @@ const Combobox = ({
             leaveTo="opacity-0"
           >
             <HeadlessComboBox.Options className="z-50 absolute mt-1 max-h-60 w-full overflow-auto rounded-md bg-white py-1 text-base shadow-lg ring-1 ring-black/5 focus:outline-none sm:text-sm">
-              {options?.map(option => (
+              {filteredOptions?.map(option => (
                 <HeadlessComboBox.Option
                   key={option}
                   value={option}
-                  className="relative cursor-pointer select-none py-2 pl-4 pr-4 text-gray-900 hover:bg-gray-100"
+                  className={optionClasses}
                 >
                   {option}
                 </HeadlessComboBox.Option>
               ))}
+              {query && !options?.includes(query) && (
```

Comment:
if `v` is falsy then `!v?.foo()` will be true, so this expression renders when `options` is falsy. I don't have a suggestion because I'm not clear on what is intended but I'll ask that the the logic be more explicit and clear, whatever is intended. Consider binding a const.

actually… why is `options` allowed to be undefined? what's a Combobox without any options? if you just change it to `options: string[]` then several lines in the component will be less shaky. When I tried this locally nothing was relying on the optionality.

Reply:
Yea good point, just made options required.

The expression checks if a query exists, and if the query is not already contained in the list of options, then gives the user the option to "add" it if so.